core/src/apps/common/cbor.py

```diff
@@ -184,10 +193,21 @@ def _cbor_decode(cbor: bytes) -> Tuple[Value, bytes]:
         return res, data
     elif fb_type == _CBOR_TAG:
         val, data = _read_length(cbor[1:], fb_aux)
-        item, data = _cbor_decode(data)
-        if val == _CBOR_RAW_TAG:  # only tag 24 (0x18) is supported
+        if val == _CBOR_RAW_TAG:
+            item, data = _cbor_decode(data)
             return item, data
+        elif val == _CBOR_SET_TAG:
+            aux = data[0] & _CBOR_INFO_BITS
+            ln, data = _read_length(data[1:], aux)
+            res = []
```

Comment:
Use standard list and the `in` keyword to deal with duplicates here, because (once again) lists aren't hashable. So I can't use a set here.